### PR TITLE
Add cross-platform image builds

### PR DESCRIPTION
**Pull request by jarojasm95**
This adds building an `arm64` flavored image - this means aladdin now has native support for the Mac M1 chipset 🥳

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -16,6 +16,9 @@ jobs:
       - name: Check Out Repo
         uses: actions/checkout@v2
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
```
Comment (jarojasm95) [inline .github/workflows/release.yaml:20]: needed for cross-platform builds

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -40,6 +43,7 @@ jobs:
         uses: docker/build-push-action@v2
         with:
           push: true
+          platforms: linux/amd64,linux/arm64
```
Comment (jarojasm95) [inline .github/workflows/release.yaml:46]: 🥳

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM python:3.8.5-buster as build
+FROM python:3.8.12-buster as build
```
Comment (jarojasm95) [inline Dockerfile:1]: opportunistic upgrade

File: Dockerfile [hunk 4]
```diff
@@ -40,8 +40,8 @@ RUN pip install --no-cache-dir pip==20.2.3
 
 # Update all needed tool versions here
 
-ARG AWS_IAM_AUTHENTICATOR_VERSION=1.17.9
-RUN curl -o /usr/local/bin/aws-iam-authenticator https://amazon-eks.s3.us-west-2.amazonaws.com/$AWS_IAM_AUTHENTICATOR_VERSION/2020-08-04/bin/linux/amd64/aws-iam-authenticator && \
+ARG AWS_IAM_AUTHENTICATOR_VERSION=1.21.2
```
Comment (jarojasm95) [inline Dockerfile:43]: same here - small opportunistic upgrade

File: Dockerfile [hunk 5]
```diff
@@ -50,33 +50,26 @@ RUN curl -L -o- https://download.docker.com/linux/static/stable/x86_64/docker-$D
     chmod 755 /usr/local/bin/docker
 
 ARG KUBE_VERSION=1.19.7
-RUN curl -L -o /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v$KUBE_VERSION/bin/linux/amd64/kubectl && \
+RUN curl -L -o /usr/local/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v$KUBE_VERSION/bin/linux/$(dpkg --print-architecture)/kubectl && \
     chmod 755 /usr/local/bin/kubectl
 
 ARG HELM_VERSION=3.5.2
```
Comment (jarojasm95) [inline Dockerfile:56]: I wanted to upgrade this to 3.6 but we've gotten bit before by helm upgrades - so did not put that here

File: Dockerfile [hunk 6]
```diff
@@ -50,33 +50,26 @@ RUN curl -L -o- https://download.docker.com/linux/static/stable/x86_64/docker-$D
...
+RUN curl -Lo kops https://github.com/kubernetes/kops/releases/download/v$KOPS_VERSION/kops-linux-$(dpkg --print-architecture) && \
     chmod +x ./kops && \
     mv ./kops /usr/local/bin/
 
 ARG ISTIO_VERSION=1.9.2
 RUN curl -L https://istio.io/downloadIstio | ISTIO_VERSION="$ISTIO_VERSION" sh - && \
     mv /istio-$ISTIO_VERSION/bin/istioctl /usr/local/bin/istioctl
 
-ARG K3D_VERSION=4.4.4
+ARG K3D_VERSION=4.4.8
```
Comment (jarojasm95) [inline Dockerfile:70]: minor patch version seemed ok

File: Dockerfile [hunk 7]
```diff
@@ -50,33 +50,26 @@ RUN curl -L -o- https://download.docker.com/linux/static/stable/x86_64/docker-$D
...
 RUN curl -s https://raw.githubusercontent.com/rancher/k3d/main/install.sh | TAG=v$K3D_VERSION bash
 
-# Install edgectl
-RUN curl -fL https://metriton.datawire.io/downloads/linux/edgectl -o /usr/local/bin/edgectl && \
-    chmod a+x /usr/local/bin/edgectl
```
Comment (jarojasm95) [inline Dockerfile:75]: this is deprecated and also does not have `arm64` support 😞

Comment (jarojasm95) [reply Dockerfile:75]: have not found a specific announcement for this, only this commit - https://github.com/emissary-ingress/emissary/commit/01b1f4eeabf1022b725198431044bf8455015e33

**Review by abandali — APPROVED** [review]
LGTM

**Pull request merged**